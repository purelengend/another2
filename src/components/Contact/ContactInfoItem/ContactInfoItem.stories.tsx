import type { Meta, StoryObj } from '@storybook/react'
import ContactInfoItem from './index'
import locationIcon from '@assets/icons/Location_Icon.svg'

const meta = {
  title: 'Practice Two/ContactInfoItem Component',
  component: ContactInfoItem,
  decorators: [
    Story => (
      <div
        style={{
          backgroundColor: '#ECB203',
          width: '100vw',
          height: '100%',
          padding: '20px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof ContactInfoItem>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {
  args: {
    icon: locationIcon,
    title: 'our main office',
    content: 'SoHo 94 Broadway St New York, NY 1001',
  },
}
