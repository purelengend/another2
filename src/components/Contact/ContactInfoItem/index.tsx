import React, { memo } from 'react'
import { Flex, Box, Image, Text } from '@chakra-ui/react'

interface ContactInfoItemProps {
  icon: string
  title: string
  content: JSX.Element | string
}

const ContactInfoItem: React.FC<ContactInfoItemProps> = ({
  icon,
  title,
  content,
}) => (
  <Box
    display="flex"
    flexDirection="column"
    backgroundColor="background"
    padding="30px"
    width={{
      base: '300px',
      sm: '340px',
      md: '350px',
      lg: '220px',
      xl: '270px',
    }}
    margin="0 auto"
    textAlign="center"
    justifyContent="flex-start"
    alignItems="center"
  >
    <Flex flexDirection="column" justifyContent="center" alignItems="center">
      <Image src={icon} alt={`${title} icon`} boxSize="64px" />
      <Text
        textTransform="uppercase"
        margin={5}
        fontFamily="OpenSans-Bold"
        fontSize="18px"
      >
        {title}
      </Text>
    </Flex>
    {content}
  </Box>
)

export default memo(ContactInfoItem)
