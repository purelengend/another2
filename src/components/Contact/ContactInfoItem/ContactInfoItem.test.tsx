import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ContactInfoItem from '.'

describe('ContactInfoItem component', () => {
  const props = {
    icon: 'icon-url',
    title: 'Contact Title',
    content: 'Contact Content',
  }

  it('renders correctly', () => {
    const { container, getByAltText, getByText } = render(
      <ContactInfoItem {...props} />
    )

    const iconElement = getByAltText(`${props.title} icon`)
    const titleElement = getByText(props.title)
    const contentElement = getByText(props.content)

    expect(iconElement).toBeInTheDocument()
    expect(titleElement).toBeInTheDocument()
    expect(contentElement).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
