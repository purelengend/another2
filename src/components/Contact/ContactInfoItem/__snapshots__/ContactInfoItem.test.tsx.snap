// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoItem component renders correctly 1`] = `
<div>
  <div
    class="css-1gfsowa"
  >
    <div
      class="css-wkpw2c"
    >
      <img
        alt="Contact Title icon"
        class="chakra-image css-1wsuzl7"
        src="icon-url"
      />
      <p
        class="chakra-text css-1hjuipy"
      >
        Contact Title
      </p>
    </div>
    Contact Content
  </div>
</div>
`;
